x/stakeibc/keeper/icacallbacks.go

```diff
@@ -116,3 +120,60 @@ func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack
 	k.RecordsKeeper.RemoveDepositRecord(ctx, cast.ToUint64(recordId))
 	return nil
 }
+
+// ----------------------------------- Reinvest Callback ----------------------------------- //
+func (k Keeper) MarshalReinvestCallbackArgs(ctx sdk.Context, delegateCallback types.ReinvestCallback) ([]byte, error) {
+	out, err := proto.Marshal(&delegateCallback)
+	if err != nil {
+		k.Logger(ctx).Error(fmt.Sprintf("MarshalReinvestCallbackArgs %v", err.Error()))
+		return nil, err
+	}
+	return out, nil
+}
+
+func (k Keeper) UnmarshalReinvestCallbackArgs(ctx sdk.Context, delegateCallback []byte) (*types.ReinvestCallback, error) {
+	unmarshalledReinvestCallback := types.ReinvestCallback{}
+	if err := proto.Unmarshal(delegateCallback, &unmarshalledReinvestCallback); err != nil {
+        k.Logger(ctx).Error(fmt.Sprintf("UnmarshalReinvestCallbackArgs %v", err.Error()))
+		return nil, err
+	}
+	return &unmarshalledReinvestCallback, nil
+}
+
+func ReinvestCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack *channeltypes.Acknowledgement_Result, args []byte) error {
+	// invariant: sendMsg.FromAddress == withdrawalAddress && sendMsg.ToAddress == delegationAddress
+	k.Logger(ctx).Info("ReinvestCallback executing", "packet", packet)
+
+	if ack == nil {
+		// transaction on the host chain failed
+		// don't create a record
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "ack is nil")
```

Comment:
Is there any error information in the packet that we could log here?

Reply:
logging the packet can't hurt! done.